### PR TITLE
Anon comments

### PR DESCRIPTION
**Pull request by ethanmasters**
The anonymous comment feature from an acceptance criterion in Sprint 2 was big enough to warrant an entire ticket for Sprint 3. This adds the feature, admin edge cases, and spec tests.

### CODE REVIEW

File: app/views/comments/_comment.html.erb [hunk 1]
```diff
@@ -5,7 +5,14 @@
     <br>
```
Comment (veritas4) [inline app/views/comments/_comment.html.erb:5]: If the current user is the author of the comment, they will not see their own name on the comment?

File: app/views/comments/_comment.html.erb [hunk 2]
```diff
@@ -5,7 +5,14 @@
...
+        Anonymous
+      <% else %>
+        <%= comment.user.name %>
+        <% if comment.anonymous %>
+          (Posted as Anonymous)
+        <% end %>
+      <% end %>
     </p>
 
     <p>
```
Comment (veritas4) [inline app/views/comments/_comment.html.erb:18]: The current user won't be see their name under the comment if they're the author of the anonymous comment?

**Review by veritas4 — COMMENTED** [review]
Great test cases for your feature. 
Once you clarify my previous question, the pull request should be good to merge.

File: app/views/comments/_comment.html.erb [hunk 3]
```diff
@@ -5,7 +5,14 @@
     <br>
     <p>
       <strong>User:</strong>
-      <%= comment.user.name %>
+      <% if comment.anonymous && !(current_user.admin? || current_user == comment.user) %>
+        Anonymous
```
Comment (veritas4) [inline app/views/comments/_comment.html.erb:9]: Is the author not able to see their own name under an anonymous comment that they made?

Comment (ethanmasters) [reply app/views/comments/_comment.html.erb:9]: The author will be given the same thing that admin will see. They will see their name and a message next to their name that it was posted anonymously. I will add comments in the code to explain it better.
